src/sentry/grouping/enhancer/__init__.py

```diff
@@ -474,6 +497,135 @@ def visit_quoted_ident(self, node, children):
         return node.match.groups()[0].lstrip("!")
 
 
+def _update_frames_from_cached_values(
+    frames: Sequence[dict[str, Any]], cache_key: str, platform: str, dry_run: bool = True
+) -> bool:
+    """
+    This will update the frames of the stacktrace if it's been cached.
+    Set dry_run to False to actually change the frames.
+    Returns if the merged has correctly happened.
+    """
+    frames_changed = False
+    # XXX: Test the fallback value
+    changed_frames_values = cache.get(cache_key, {})
+    # This helps tracking changes in the hit/miss ratio of the cache
+    metrics.incr(
+        "save_event.stacktrace.cache.get",
+        tags={"success": bool(changed_frames_values), "platform": platform},
+    )
+    if changed_frames_values:
+        try:
+            for frame, changed_frame_values in zip(frames, changed_frames_values):
+                if not dry_run:
```

Comment:
What's the value of the dry run here? I feel like we're not logging anything, so this really just proves that `cache.get` works